packages/graphql/src/translate/create-interface-projection-and-params.ts

```diff
@@ -206,37 +209,56 @@ function createInterfaceProjectionAndParams({
             });
         }
 
-        subquery.push(`RETURN ${projectionStr} AS ${field.fieldName}`);
-        globalParams = {
-            ...globalParams,
-            ...projectionParams,
-        };
+        const projectionSubqueryClause = CypherBuilder.concat(...projectionSubQueries);
+        return new CypherBuilder.RawCypher((env) => {
+            const subqueryStr = compileCypherIfExists(projectionSubqueryClause, env);
+            const returnStatement = `RETURN ${projectionStr} AS ${field.fieldName}`;
 
-        return subquery.join("\n");
+            return [[...subquery, subqueryStr, returnStatement].join("\n"), {}]; // TODO: pass params here instead of globalParams
+        });
     });
-    let interfaceProjection = [`WITH ${fullWithVars.join(", ")}`, "CALL {", subqueries.join("\nUNION\n"), "}"];
 
-    if (field.typeMeta.array) {
-        interfaceProjection = [
-            `WITH ${fullWithVars.join(", ")}`,
-            "CALL {",
-            ...interfaceProjection,
-            `RETURN collect(${field.fieldName}) AS ${field.fieldName}`,
-            "}",
-        ];
+    const optionsInput = resolveTree.args.options as GraphQLOptionsArg | undefined;
+    let withClause: CypherBuilder.With | undefined;
```

Comment:
Might be nicer to make `.With` return undefined directly if it's falsy.

Reply:
Mmm, I think that would be a bit of an antipattern, as we are defining "with" depending on the options, which may be unrelated to `.with`.

We may improve this a bit if we create a new function with this content and return `With | undefined` but I feel that's a bit of overdoing at this stage